### DISCUSSION TITLE
Can we store the current editing locale to session rather than memory?

### DISCUSSION THREAD

**Discussion by richardvanbergen** [payloadcms/payload#174] [Feature Requests & Ideas]
In relation to #173 could we maybe store the current locale in a cookie alongside the user session?

I've noticed that refreshing the page without the query string can result in it switching to the default language as the current language selection is stored in memory.

I have also had some feedback from editors that sometimes it forgets what language it's in and I suspect this might be the reason.

**Comment by jmikrut** [comment]
Hey @richardvanbergen — 

This is actually in the works right now, falling under a **User Preferences** feature, where we'll be saving ways that the user interacts with the Admin UI.

For example, things we'll be storing include:

- Locale
- Active filters per collection
- Collapsed / uncollapsed state of `blocks`
- Etc. moving forward

The API for saving a user preference is going to be open for use as well so if you ever need to do something similar in a custom component, you will be able to leverage our pattern.

This is slated for release within the next month. It'll hit _**before**_ our roadmap goes public. We have some things in the works right now, including a Product Hunt launch, that are taking precedence over the Roadmap launch... but many things are on their way!

**Reply by richardvanbergen** [reply]
Great I look forward to it! 

> Collapsed / uncollapsed state of blocks

🥳 We use a lot of blocks so this will be extremely helpful.

**Reply by Ontopic** [reply]
Any timeframe on this? Before I start building all kinds of redundant components 😅

**Reply by jmikrut** [reply]
Within 30 days! How's that?

**Reply by Ontopic** [reply]
Perfect! These settings will save me building a lot of components the client is at some point gonna request, but not necessary before then

**Reply by jmikrut** [reply]
Question for you both @richardvanbergen and @Ontopic —

We are evaluating saving user preferences in local storage OR saving them on the user itself in the database under a new `preferences` field.

Do either of you have a preference on where we store them?

**Reply by Ontopic** [reply]
`preferences` field +1

Editors work from home, on their phone (if we improve responsiveness here and there) and in the office. Syncing setting would be superior. If we add the ability to set access levels on the `preferences` we could even enforce certain preferences for certain users (e.g. French translator has an unchangeable filter for only seeing collections in his "preferred" language; fr)

**Reply by jmikrut** [reply]
> (if we improve responsiveness here and there) 

Also working on it 😎 

I agree. We are in camp "preferences field" as well!

**Reply by richardvanbergen** [reply]
100% with the preference field approach.

**Comment by jmikrut** [comment]
Just an update to this thread: User Preferences have been released and Payload now persists the user's last used locale to the database.

👍
